src/Nri/Ui/Table/V8.elm

```diff
@@ -0,0 +1,368 @@
+module Nri.Ui.Table.V8 exposing
+    ( Column, SortDirection(..), custom, string, rowHeader, placeholderColumn
+    , view, viewWithoutHeader
+    , viewLoading, viewLoadingWithoutHeader
+    )
+
+{-| Upgrading from V7:
+
+  - New attribute added to define `alternatingRowColors`, set it to true to get default behavior.
+  - Consider moving to `SortableTable`, as in V4 a non-interactive table renders
+    just the same but has additional flexibility in the API and will make future
+    upgrades easier.
+
+@docs Column, SortDirection, custom, string, rowHeader, placeholderColumn
+
+@docs view, viewWithoutHeader
+
+@docs viewLoading, viewLoadingWithoutHeader
+
+-}
+
+import Accessibility.Styled.Style as Style
+import Css exposing (..)
+import Css.Animations
+import Css.Global
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attributes exposing (css)
+import Nri.Ui.Colors.V1 exposing (..)
+import Nri.Ui.Fonts.V1 exposing (baseFont)
+
+
+{-| Closed representation of how to render the header and cells of a column
+in the table
+-}
+type Column data msg
+    = Column (Html msg) (data -> Html msg) Style (data -> List Style) (Maybe SortDirection) CellType Bool
+
+
+{-| Which direction is a table column sorted? Only set these on columns that
+actually have an explicit sort!
+-}
+type SortDirection
+    = Ascending
+    | Descending
+
+
+{-| Is this cell a data cell or header cell?
+-}
+type CellType
+    = RowHeaderCell
+    | DataCell
+
+
+cell : CellType -> List (Attribute msg) -> List (Html msg) -> Html msg
+cell cellType attrs =
+    case cellType of
+        RowHeaderCell ->
+            th (Attributes.scope "row" :: attrs)
+
+        DataCell ->
+            td attrs
+
+
+{-| A column that renders some aspect of a value as text
+-}
+string :
+    { header : String
+    , value : data -> String
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+string { header, value, width, cellStyles, sort } =
+    Column (Html.text header) (value >> Html.text) (Css.width width) cellStyles sort DataCell False
+
+
+{-| Creates a placeholder column which will reserve the space for the column,
+this can be used when multiple tables have similar data and we want to keep
+the size consistent.
+-}
```

Comment:
🎨 Can we document that using this attribute will make the whole table switch to table-layout: fixed?

Reply:
(Assuming we don't make the layout a separate attribute as suggested in some other comment)

Reply:
Suer!